presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -112,3 +112,37 @@ To enable SSL/TLS for Presto internal communication, do the following:
 
         internal-communication.https.keystore.path=<keystore path>
         internal-communication.https.keystore.key=<keystore password>
+
+
+Enabling encryption introduces performance impact. Performance degradation can
+vary based on the environment, queries, concurrency.
+
+For queries that do not require transferring too much data between the Presto
+nodes (e.g. `SELECT count(*) FROM table`), performance impact is negligible.
+
+However, for CPU intensive queries which require considerable amount of data
+to be transferred between the nodes (for example when repartitioning is needed:
+distributed joins, aggregations, and windowing functions), performance impact
+might be considerable. Slowdown may vary from 10% to even 100%+ depending on the network
+traffic and the CPU utilization.
+
+TLS encryption uses `/dev/urandom` system device as a source of entropy.
+This device is system-wide synchronized, thus has a limited throughput.
```

Comment:
This device has limited throughput, so on environments with high network bandwidth (e.g. InfiniBand), powerful CPUs (multi-core, high-end server class) and high concurrency, it may become a bottleneck.